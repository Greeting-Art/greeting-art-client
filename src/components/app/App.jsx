import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import ArtGenerator from '../../containers/ArtGenerator';
import ECard from '../../containers/ECard';
import Gallery from '../../containers/Gallery';
import AboutUs from '../../containers/AboutUs';
import Alchemy from '../specialized/Alchemy';
import BDRyan from '../specialized/BDRyan';
import QuackMan from '../specialized/QuackMan';
import Header from '../header/Header';
import './app.css';

export default function App() {
  return (
    <Router>
      <Header />
      <Switch>
        <Route
          path="/"
          exact
          render={(routerProps) => <ArtGenerator {...routerProps} />}
        />
        <Route
          path="/ecard"
          exact
          render={(routerProps) => <ECard {...routerProps} />}
        />
        <Route
          path="/gallery"
          exact
          render={(routerProps) => <Gallery {...routerProps} />}
        />
        <Route
          path="/aboutus"
          exact
          render={(routerProps) => <AboutUs {...routerProps} />}
        />
        <Route
          path="/alchemy"
          exact
          render={(routerProps) => <Alchemy {...routerProps} />}
        />
        <Route
          path="/ryan"
          exact
          render={(routerProps) => <BDRyan {...routerProps} />}
        />
        <Route
          path="/quack"
          exact
          render={(routerProps) => <QuackMan {...routerProps} />}
        />
      </Switch>
    </Router>
  );
}
