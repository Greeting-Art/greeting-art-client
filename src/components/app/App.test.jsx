/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

jest.mock('../../containers/ArtGenerator', () => () => <div>hey</div>);

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
