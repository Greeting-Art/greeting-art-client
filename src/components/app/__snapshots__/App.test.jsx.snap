// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <section
    class="fullHeader"
  >
    <div
      class="linkBar"
    >
      <button
        aria-controls="simple-menu"
        aria-haspopup="true"
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root iconButtonMUI MuiIconButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <a
      href="/"
    >
      <div
        class="titleBar"
      >
        <figure>
          <img
            src="/path/image"
            width="640"
          />
        </figure>
      </div>
    </a>
    <div
      class="rightHelpBar"
    >
      <button
        aria-controls="help-menu"
        aria-haspopup="true"
        aria-label="help"
        class="MuiButtonBase-root MuiIconButton-root iconButtonMUI MuiIconButton-colorInherit"
        id="helpButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3c-.46 0-.93.04-1.4.14-2.76.53-4.96 2.76-5.48 5.52-.48 2.61.48 5.01 2.22 6.56.43.38.66.91.66 1.47V19c0 1.1.9 2 2 2h.28c.35.6.98 1 1.72 1s1.38-.4 1.72-1H14c1.1 0 2-.9 2-2v-2.31c0-.55.22-1.09.64-1.46C18.09 13.95 19 12.08 19 10c0-3.87-3.13-7-7-7zm2 16h-4v-1h4v1zm0-2h-4v-1h4v1zm-1.5-5.59V14h-1v-2.59L9.67 9.59l.71-.71L12 10.5l1.62-1.62.71.71-1.83 1.82z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </section>
  <div>
    hey
  </div>
</DocumentFragment>
`;
